fractions/jose/src/main/java/org/wildfly/swarm/jose/JoseConfiguration.java

```diff
@@ -0,0 +1,260 @@
+        signatureDataEncoding.set(encoding);
+        return this;
+    }
+
+    public boolean signatureDataEncoding() {
+        return this.signatureDataEncoding.get();
+    }
+
+    public JoseConfiguration signatureDataDetached(boolean detached) {
+        signatureDataDetached.set(detached);
+        return this;
+    }
+
+    public boolean signatureDataDetached() {
+        return this.signatureDataDetached.get();
+    }
+
+    public JoseFormat signatureFormat() {
+        return JoseFormat.valueOf(this.signatureFormat.get());
+    }
+
+    public JoseConfiguration encryptionFormat(JoseFormat format) {
+        encryptionFormat.set(format.name());
+        return this;
+    }
+
+    public JoseFormat encryptionFormat() {
+        return JoseFormat.valueOf(this.encryptionFormat.get());
+    }
+
+    public JoseConfiguration encryptionKeyAlias(String keyAlias) {
+        this.encryptionKeyAlias.set(keyAlias);
+        return this;
+    }
+
+    public String encryptionKeyAlias() {
+        return this.encryptionKeyAlias.get();
+    }
+
+    public JoseConfiguration encryptionKeyPassword(String password) {
+        this.encryptionKeyPassword.set(password);
+        return this;
+    }
+
+    public String encryptionKeyPassword() {
+        return this.encryptionKeyPassword.get();
+    }
+
+    public JoseConfiguration keyEncryptionAlgorithm(String algorithm) {
+        keyEncryptionAlgorithm.set(algorithm);
+        return this;
+    }
+
+    public String keyEncryptionAlgorithm() {
+        return this.keyEncryptionAlgorithm.get();
+    }
+
+    public JoseConfiguration contentEncryptionAlgorithm(String algorithm) {
+        contentEncryptionAlgorithm.set(algorithm);
+        return this;
+    }
+
+    public String contentEncryptionAlgorithm() {
+        return this.contentEncryptionAlgorithm.get();
+    }
+    /**
+     * Keystore type.
+     */
+    @Configurable("swarm.jose.keystore.type")
+    @AttributeDocumentation("Keystore type")
+    private Defaultable<String> keystoreType = string(DEFAULT_KEYSTORE_TYPE);
+
+    /**
+     * Path to the keystore.
+     */
+    @Configurable("swarm.jose.keystore.path")
+    @AttributeDocumentation("Path to the keystore")
+    private Defaultable<String> keystorePath = string(DEFAULT_KEYSTORE_PATH);
+
+    /**
+     * Password for the keystore.
+     */
+    @Configurable("swarm.jose.keystore.password")
+    @AttributeDocumentation("Password to the keystore")
+    private Defaultable<String> keystorePassword = string(DEFAULT_KEYSTORE_PASSWORD);
+
+    /**
+     * Signature algorithm.
+     */
+    @Configurable("swarm.jose.signature.algorithm")
+    @AttributeDocumentation("Signature algorithm")
+    private Defaultable<String> signatureAlgorithm = string(DEFAULT_SIGNATURE_ALGORITHM);
+
+    /**
+     * Signature Format.
+     */
+    @Configurable("swarm.jose.signature.format")
+    @AttributeDocumentation("Compact or JSON JWS format, support for JSON is optional")
```

Comment:
What are the possible values?

Reply:
Also, not quite sure as a user what "support for JSON is optional" means. What guarantees do I have?

Reply:
I'll update the docs for this and other attributes.
I've added the 'optional' note to let Jose4J be used which does not support it. I guess it is like with some of other Java APIs which are optional for the implementers. Or like (async) CompletionCallback. This is why I'm sticking with CXF for now at least as the default impl given that it supports it

Reply:
It should work for the users who may not need JSON format. I'm not sure yet how to express the fact the default impl guarantees it, given no guarantee exists CXF will remain as a default one, I guess we'll do it right at some point :-).